hypothesis-python/RELEASE.rst

```diff
@@ -0,0 +1,6 @@
+RELEASE_TYPE: minor
+
+This release adds a new mechanism to infer strategies for classes
+defined using :pypi:`attrs`, based on the the type, converter, or
+validator of each attribute.  This inference is now built in to
+:func:`~hypothesis.strategies.builds` and :func:`~hypothesis.strategies.from_type`.
```

Comment:
Is it worth calling out that this will change the behaviour of strategies using `builds` for attrs where there's a default type they're omitting?

e.g. if I have

```python
@attr.s()
class Foo(object):
    bar = attr.ib(default=0, converter=int)
```

Then previously this would have generated only `Foo(0)`, but now can generate `Foo(n)` for any `n`.

(I think it's fine to change this, I just think it's worth explicitly mentioning)

Reply:
It will only change the behaviour in this case if you call `builds(Foo, bar=infer)`; or succeed instead of failing if you try to build something with a required argument but no strategy.

The inference about the default value is so that inferred strategies can "shrink away" to the default value, and then `None` if that would be valid.

Reply:
Ah, right! Sorry I was forgetting how defaults and inference interacted.

Reply:
No worries, it's the opposite way for Django models thanks to an odd early decision!